GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java

```diff
@@ -124,10 +121,16 @@ public static synchronized void incrementCount(QuestionAnswerStore answer,
      */
     @SuppressWarnings("unchecked")
     public List<SurveyQuestionSummary> listByResponse(String questionId, String questionResponse) {
+        //Reject too-long key; TODO truncate instead?
+        String answer = questionResponse;
+        if (answer.length() > Constants.MAX_LENGTH) {
```

Comment:
Just like previously do you want to check whether answer is null?

Reply:
ok